modules/memberships/class-jetpack-memberships.php

```diff
@@ -250,16 +250,19 @@ public function render_button( $attrs ) {
 		}
 		$button_styles = implode( ';', $button_styles );
 		add_thickbox();
+		global $wp;
 		return sprintf(
-			'<div class="wp-block-button %1$s"><a role="button" data-blog-id="%2$d" data-powered-text="%3$s" data-plan-id="%4$d" data-lang="%5$s" class="%6$s" style="%7$s">%8$s</a></div>',
+			'<div class="wp-block-button %1$s"><a role="button" %10$s href="https://subscribe.wordpress.com/memberships/?blog=%2$d&plan=%4$d&lang=%5$s&pid=%9$d&redirect=%3$s" class="%6$s" style="%7$s">%8$s</a></div>',
 			esc_attr( Jetpack_Gutenberg::block_classes( self::$button_block_name, $attrs ) ),
 			esc_attr( $data['blog_id'] ),
-			esc_attr( $data['powered_text'] ),
+			esc_attr( rawurlencode( home_url( $wp->request ) ) ), // Needed for redirect back in case of redirect-based flow.
 			esc_attr( $data['id'] ),
```

Comment:
Now that we have both the post ID and the blog ID in there, maybe we should consider renaming this into `plan_id` to avoid any confusion?

Reply:
That ship has sailed, we have code using this as a shortcode in other codebases and referring to it as `id`. Changing this now will break that codebase.

Reply:
Done!